src/components/tvproviders/schedulesdirect.js

```diff
@@ -83,13 +83,9 @@ export default function (page, providerId, options) {
         loading.hide();
     }
 
-    function sha256(str) {
-        if (!self.TextEncoder) {
```

Comment:
This should probably be
```
if (!window.TextEncoder)
```

The goal being to test if TextEncoder is supported or not.

Reply:
Although, theoretically, this would only ever be executed on desktop or mobile, since TVs don't have access to the dashboard. And these always have TextEncoder available, so not sure if it's needed or not...

Reply:
Just safe to remove the hashing client side and always do the it on the server then?

Reply:
Yeah, removing the hashing here and doing it on the server, you can remove all of this and just send the unhashed password :)